app/src/forms/auth/middleware/userAccess.js

```diff
@@ -43,54 +39,12 @@ const currentUser = async (req, res, next) => {
   return setUser(req, res, next);
 };
 
-// Temporary version of currentUser that only calls sets req.currentUser for
-// requests using Bearer tokens. This can only be used for routes that call
-// controller methods that don't use req.currentUser.
-//
-// Plan: If this code is a success it will eventually be rolled out to all
-// endpoints that don't need req.currentUser for API Key use. Then, all those
-// that do need req.currentUser will be refactored to use the minimal query we
-// need.
-const currentUserTemp = async (req, res, next) => {
-  // Check if authorization header is a bearer token
-  if (req.headers && req.headers.authorization && req.headers.authorization.startsWith('Bearer ')) {
-    // need to check keycloak, ensure the bearer token is valid
-    const token = req.headers.authorization.substring(7);
-    const ok = await keycloak.grantManager.validateAccessToken(token);
-    if (!ok) {
-      return new Problem(403, { detail: 'Authorization token is invalid.' }).send(res);
-    }
-  }
-
-  // Temporarily set a flag to prevent expensive database calls. We will
-  // eventually move all routes to working in this way, and then the extra
-  // downstream logic can be removed.
-  req.chefsLoadForms = false;
-
-  return setUser(req, res, next);
-};
-
-// To deal with performance problems, we are going to move away from setting the
-// req.currentUser.forms and req.currentUser.deletedForms data. The code using
-// those will eventually be removed from this function.
 const _getForm = async (currentUser, formId) => {
-  let forms;
-  if (currentUser.forms) {
-    forms = currentUser.forms;
-  } else {
-    forms = await service.getUserForms(currentUser, { active: true, formId: formId });
-  }
-
+  const forms = await service.getUserForms(currentUser, { active: true, formId: formId });
   let form = forms.find((f) => f.formId === formId);
```

Comment:
There's a few places now where we get the forms from the service and then filter out the one we want. This second step is _probably_ not needed, but we'd need additional checks around the response from the service. I'm going to leave it for now so that it's an easier review, but it's pretty ugly code so I'll clean it up once I put some tests in place.